include/macroni/Analysis/Kernel/RCUAnalysis.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <mlir/IR/Operation.h>
+
+namespace macroni::kernel {
+struct kernel_analysis {
+  kernel_analysis(mlir::Operation *op);
```

Comment:
Add hooks here for reporting - all those functions that currently report to `errs` or any other stream should instead call the hook and the user can decide what to do (and the output to standard streams can be the default for example).